common/defs.go

```diff
@@ -197,58 +199,74 @@ func (p *Policy) Validate() error {
 				}
 			}
 			if len(badRanges) > 0 {
-				errMsg += fmt.Sprintf("The following port ranges are invalid: %s. ", strings.Join(badRanges, ", "))
+				errMsg = append(errMsg, fmt.Sprintf("Rule #%d: The following port ranges are invalid: %s.", ruleNo, strings.Join(badRanges, ", ")))
 			}
 			badPorts := make([]string, 0)
 			for _, port := range r.Ports {
 				if port < 0 || port > MaxPortNumber {
-					badPorts = append(badPorts, string(port))
+					badPorts = append(badPorts, fmt.Sprintf("%d", port))
 				}
 			}
 			if len(badPorts) > 0 {
-				errMsg += fmt.Sprintf("The following ports are invalid: %s. ", strings.Join(badPorts, ", "))
+				errMsg = append(errMsg, fmt.Sprintf("Rule #%d: The following ports are invalid: %s.", ruleNo, strings.Join(badPorts, ", ")))
 			}
 		}
-		if r.Protocol == "icmp" {
+		if r.Protocol != "icmp" {
```

Comment:
Could we turn this around? Have the check for ICMP first? And then have an 'else' for the 'invalid protocol' branch?